plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSFlowgraphValidator.java

```diff
@@ -88,191 +78,52 @@ private Void checkCancelled() {
 	}
 
 	/**
-	 * Checks all flow graph related validations
+	 * Triggers all flow graph related validations
 	 */
 	@Check
 	public void checkFlowGraphs(Script script) {
 		// Note: The Flow Graph is NOT stored in the meta info cache. Hence, it is created here at use site.
 		// In case the its creation is moved to the N4JSPostProcessor, care about an increase in memory consumption.
 		N4JSFlowAnalyser flowAnalyzer = new N4JSFlowAnalyser(this::checkCancelled);
 
-		DeadCodeAnalyser dcv = new DeadCodeAnalyser();
-		UsedBeforeDeclaredAnalyser cvgv1 = new UsedBeforeDeclaredAnalyser();
-		NullDereferenceAnalyser nda = new NullDereferenceAnalyser();
+		FlowValidator[] fValidators = {
+				new DeadCodeValidator(keywordProvider),
+				new UsedBeforeDeclaredValidator(),
+				new NullUndefinedValidator(n4jsCore, findReferenceHelper),
+				new MissingReturnOrThrowValidator(typeSystem, jsVariantHelper) };
```

Comment:
instead of this pattern, please use DI for consistency (i.e. DeadCodeValidator, etc. are injected and inject themselves what they need).

Reply:
👍